src/test/scala/mesosphere/marathon/api/akkahttp/v2/AppsControllerTest.scala

```diff
@@ -0,0 +1,1433 @@
+           |  "container": {
+           |    "type": "${`type`}",
+           | $docker
+           |    $volumes
+           |  }
+           |}
+      """.stripMargin
+
+      When("The request is processed")
+      HttpEntity(body).withContentType(ContentTypes.`application/json`)
+    }
+  }
+
+  case class FixtureWithRealGroupManager(
+      initialRoot: RootGroup = RootGroup.empty,
+      clock: SettableClock = new SettableClock(),
+      auth: TestAuthFixture = new TestAuthFixture {
+        override val UnauthorizedStatus = 403
+        override val NotAuthenticatedStatus = 401
+      },
+      service: MarathonSchedulerService = mock[MarathonSchedulerService],
+      appInfoService: AppInfoService = mock[AppInfoService],
+      healthCheckManager: HealthCheckManager = mock[HealthCheckManager],
+      instanceTracker: InstanceTracker = mock[InstanceTracker],
+      taskKiller: TaskKiller = mock[TaskKiller],
+      configArgs: Seq[String] = Seq("--enable_features", "external_volumes")) {
+    val groupManagerFixture: TestGroupManagerFixture = new TestGroupManagerFixture(initialRoot = initialRoot)
+    val groupManager: GroupManager = groupManagerFixture.groupManager
+    val groupRepository: GroupRepository = groupManagerFixture.groupRepository
+
+    implicit val authenticator: Authenticator = auth.auth
+    implicit val authorizer: Authorizer = auth.auth
+
+    implicit val electionService: ElectionService = mock[ElectionService]
+    when(electionService.isLeader).thenReturn(true)
+
+    val config: AllConf = AllConf.withTestConfig(configArgs: _*)
+
+    val appsController = new AppsController(
+      clock = clock,
+      eventBus = system.eventStream,
+      marathonSchedulerService = service,
+      appInfoService = appInfoService,
+      healthCheckManager = healthCheckManager,
+      instanceTracker = instanceTracker,
+      taskKiller = taskKiller,
+      config = config,
+      groupManager = groupManager,
+      pluginManager = PluginManager.None
+    )
+
+    implicit val rejectionHandler: RejectionHandler = AkkaHttpMarathonService.rejectionHandler
+    val route: Route = Route.seal(appsController.route)
+
+  }
+
+  "Apps Controller" should {
+    "Create a new app successfully" in new Fixture {
+      Given("An app and group")
+      val app = App(id = "/app", cmd = Some("cmd"))
+      val (body, plan) = prepareApp(app, groupManager)
+
+      When("The create request is made")
+      clock += 5.seconds
+
+      val entity = HttpEntity(body).withContentType(ContentTypes.`application/json`)
+
+      Post(Uri./, entity) ~> route ~> check {
+        Then("It is successful")
+        status shouldEqual StatusCodes.Created
+        header[Headers.`Marathon-Deployment-Id`] should not be 'empty
+
+        And("the JSON is as expected, including a newly generated version")
+        import mesosphere.marathon.api.v2.json.Formats._
+        val expected = AppInfo(
```

Comment:
How about using a JSON string here? What do you think? This avoids serialization

Reply:
I don't know what's wrong with serialization, but yes, I can do it.

Reply:
It seems it's not possible because of the dynamic id generation.

Reply:
Just used string interpolation :).